src/core/webidl.js

```diff
@@ -139,8 +139,10 @@ function defineIdlName(escaped, data, parent) {
      data-lt="default toJSON operation">${escaped}</a>`;
   }
   if (!data.partial) {
-    return hyperHTML`<dfn data-export data-dfn-type="${linkType}" data-dfn-for="${parent &&
+    const dfn = hyperHTML`<dfn data-export data-dfn-type="${linkType}" data-dfn-for="${parent &&
       parent.name}">${escaped}</dfn>`;
+    decorateDfn(dfn, data, parentName, name);
```

Comment:
Maybe we could remove `decorateDfn` call in `findDfn` because we will always decorate dfns regardless of existence of links.

Reply:
yeah... and it's a bit less ugly... "finding" things shouldn't have side effects.... like ids being added. Long wanted to change that.

Reply:
Argh... actually, looking at the code, that change is a bit more involved than I first thought. Might need to do that as a new PR.

Reply:
Filed: https://github.com/w3c/respec/issues/2444